### PR TITLE
Auto capitalization toggle

### PR DESCRIPTION
**Pull request by HasanMansoor4**
# Description 📣

New feature related to issue #276 Allow the user to toggle off/on the Auto Capitalization of secrets keys, by having a checkbox in the settings. I would love some feedback as this is my very first open source contribution. Thank you!

# Screenhot
Feb-03-2023 04-51-47

### CODE REVIEW

**Comment by mv-turtle** [comment]
I think this is awesome!! It makes a lot of sense! Just two minor things: 
1. We moved settings to a new structure (it is now componerized across different files). Do you think you would be able to update it? If not, I could also try to do that :) Let me know!
2. I assume `default: true` is going to make it work for old users too right? @dangtony98 would you be able to have a look at the backend? I think you have much more experience, so would love to hear if you think everything's ok

**Comment by HasanMansoor4** [comment]
> 1. We moved settings to a new structure (it is now componerized across different files). Do you think you would be able to update it? If not, I could also try to do that :) Let me know!

Yeah I just noticed the merge conflicts just now lol. I'm not that experienced with React/Next.js tbh I'll look into it and keep you updated. 

> 2. I assume `default: true` is going to make it work for old users too right? @dangtony98 would you be able to have a look at the backend? I think you have much more experience, so would love to hear if you think everything's ok 

The auto capitalization boolean isn't backfilled for previous documents, but im using the ?? operator to default to true for undefined values. is this okay? 

I'd love some feedback from @dangtony98

**Comment by HasanMansoor4** [comment]
Hi @mv-turtle and @dangtony98, so I got the component refactored and its now using react query, I also removed the save button as I don't think there's a need for it.  Would love some reviews.  Thank you!

Feb-05-2023 06-31-17

File: backend/src/controllers/v2/workspaceController.ts [hunk 1]
```diff
@@ -467,4 +467,42 @@ export const deleteWorkspaceMembership = async (req: Request, res: Response) =>
...
\ No newline at end of file
+}
+
+/**
+ * Change autoCapitilzation Rule of workspace
+ * @param req
+ * @param res
+ * @returns
+ */
+export const toggleAutoCapitilzation = async (req: Request, res: Response) => {
```
Comment (dangtony98) [inline backend/src/controllers/v2/workspaceController.ts:478]: Can you change this to `toggleAutoCapitalization`? - Sorry a little nit-picky about spelling haha.

File: backend/src/routes/v2/workspace.ts [hunk 2]
```diff
@@ -118,4 +118,19 @@ router.delete( // TODO - rewire dashboard to this route
 	workspaceController.deleteWorkspaceMembership
 );
 
+
+router.post(
+	'/:workspaceId/settings',
```
Comment (dangtony98) [inline backend/src/routes/v2/workspace.ts:123]: Can you change this endpoint to PATCH `/:workspaceId/auto-capitalization` instead to be more descriptive and reflect that it's an update to a resource as opposed to creating a new resource?

**Comment by dangtony98** [comment]
Hey @HasanMansoor4!

Sorry for taking some time to review this ... I've left 2 comments on the backend side of things; just some small endpoint-naming change and fixing spelling for the controller function.

Other than that, it looks great to me!

**Comment by HasanMansoor4** [comment]
Thank you @dangtony98. Yeah all your comments make sense, I have them fixed now.

**Review by mv-turtle — APPROVED** [review]
This looks really great! Thank you so much for contributing @HasanMansoor4  :)

**Pull request merged**